platform/engineering/design-docs/2020-08-17-mobile-api-design-doc.md

```diff
@@ -0,0 +1,173 @@
+# Mobile API Design Doc
+
+**Author(s):** Jonathan Julian (CTO Support, Engineering), Alastair Dawson (CTO Support, Engineering)  
+**Last Updated:** August 17, 2020  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** Andrew Gunsch, John Paul Ashenfelter, Michael Fleet, Johnny Holton  
+
+## Overview
+
+### Objective
+The new flagship VA Mobile app contains functionality that is available on VA.gov, as well as some novel features. 
+The app needs to be powered by an API providing authenticated Veteran data. This API should:
+
+- Reuse backend integrations wherever possible
+- Not add complexity to an already complex system
+- Be able to be built and maintained without undue overhead
+- Complement and even bolster adjacent projects such as VA.gov web and Lighthouse
+
+This document is intended for engineers and engineering leadership alike.
+
+### Background
+In the winter/spring of 2020, a “VA Mobile App Discovery Sprint” was held to determine the feasibility of a flagship 
+mobile app for the VA. It was immediately apparent that the authenticated Veteran experience would provide the most value, 
+similar to the VA.gov experience. The goal of the mobile app is not to replicate every bit of VA.gov, but to provide a 
+new mode of communication to allow Veterans to interact more easily with the department. Since VA.gov is powered by 
+vets-api and VSP, it seemed like a natural fit for the mobile app to be powered by APIs in the same space.
+
+### High Level Design
+Since vets-api already contains integrations to backend services that the mobile app will need 
+(such as MVI, Claims, Appointments), the mobile API will be built inside of the vets-api codebase, and exposed in the same 
+ way that VA.gov API endpoints are exposed.
+
+The mobile API will provide functionality that is currently exclusive to VA.gov as well as some already-public functionality 
+from Lighthouse, so existing integrations will be reused. A Rails engine will house the mobile API surface area, with necessary 
```

Comment:
> A Rails engine will house the mobile API surface area

Does this mean all calls from the mobile client will be made to endpoints in the `mobile` Rails engine?

Reply:
yes